plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -3932,9 +4013,53 @@ def update_fabric_devices(self, fabric_devices):
                 self.log("Desired fabric device '{ip}' details (want): {requested_state}"
                          .format(ip=device_ip, requested_state=want_device_details), "DEBUG")
                 try:
-                    payload = {"payload": [want_device_details]}
-                    task_name = "add_fabric_devices"
-                    task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+                    device_roles = want_device_details.get("deviceRoles")
+                    if device_roles == ["CONTROL_PLANE_NODE"]:
+                        route_distribution_protocol = item.get("route_distribution_protocol")
+                        if route_distribution_protocol is None:
+                            route_distribution_protocol = "LISP_BGP"
+                        else:
+                            route_distribution_protocol_list = ["LISP_BGP", "LISP_PUB_SUB"]
+                            if route_distribution_protocol not in route_distribution_protocol_list:
+                                self.msg = (
+                                    "The 'route_distribution_protocol_list' should be in the list {protocol_list}."
+                                    .format(protocol_list=route_distribution_protocol_list)
+                                )
+                                self.status = "failed"
+                                return self
+
+                        payload = {
+                            "payload": {
+                                "deviceManagementIpAddress": device_ip,
+                                "siteNameHierarchy": fabric_name,
+                                "routeDistributionProtocol": route_distribution_protocol
+                            }
+                        }
+                        task_name = "add_control_plane_device"
+
```

Comment:
self.log(
        "Payload prepared for '{task_name}' API call: {payload}".format(
            task_name=task_name, payload=payload
        ), "DEBUG"
    )